code/modules/antagonists/roundstart_special/special_antagonist.dm

```diff
@@ -0,0 +1,129 @@
+#define SPAWNTYPE_ROUNDSTART "roundstart"
+#define SPAWNTYPE_MIDROUND "midround"
+#define SPAWNTYPE_EITHER "either"
+
+////////////////////////////////
+//////Special Role 'Controller'///////
+////////////////////////////////
+
+/datum/special_role
+	var/attached_antag_datum = /datum/antagonist/special
+	var/spawn_mode = SPAWNTYPE_ROUNDSTART
+	var/probability = 0				//The probability of any spawning
+	var/min_players = 0				//Min player count (Forgot about this, might be important)
+	var/max_players = -1			//Max player count (-1 for no max count)
+	var/proportion = 0				//The prbability per person of rolling it
+	var/max_amount = 0				//The maximum amount
+	var/role_name = "special role"
+	//----Required for roundspawn----
+	var/allowAntagTargets = FALSE	//Not used in events
+	var/latejoin_allowed = TRUE		//Can latejoins be assigned to this? If you want this to be a midround spawn, put these in the round_event
+	var/list/protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Head of Personnel", "Chief Medical Officer", "Chief Engineer", "Research Director", "Captain", "Brig Physician")
+	//----Required for midround----
+	var/event_typepath = /datum/round_event/create_special_antag
+	var/weight = 10
+	var/earliest_start = 20 MINUTES
+	var/max_occurrences = 1
+	var/holidayID = ""
+	//Preferences
+	var/preference_type = ROLE_SPECIAL
+
+/datum/special_role/New()
+	. = ..()
+	if(spawn_mode == SPAWNTYPE_ROUNDSTART)
+		return
+	//Create a new event for spawning the antag
+	var/datum/round_event_control/E = new()
+	E.name = role_name
+	E.typepath = event_typepath
+	E.weight = weight
+	E.holidayID = holidayID
+	if(config)
+		E.earliest_start = CEILING(earliest_start * CONFIG_GET(number/events_min_time_mul), 1)
+		E.min_players = CEILING(min_players * CONFIG_GET(number/events_min_players_mul), 1)
+	else
+		E.earliest_start = earliest_start
+		E.min_players = min_players
+	//Shove our event into the subsystem pool :)
+	SSevents.control += E
+
+/datum/special_role/proc/add_antag_status_to(var/datum/mind/M)
+	M.special_role = role_name
+	var/datum/antagonist/special/A = M.add_antag_datum(new attached_antag_datum())
+	A.forge_objectives(M)
+	A.equip()
+	return(A)
+
+////////////////////////////////
+//////    Round event    ///////
+////////////////////////////////
+//The event that is associated with the special antag
+//If you are making a midround event that uses this system
+//Create a datum that implements /datum/round_event/create_special_antag
+//and put that datum as the event_typepath variable in
+//The controller.
+//If your role cannot spawn midround, don't bother making it.
+
+/datum/round_event/create_special_antag
+	fakeable = FALSE
+	var/role_name
+	var/antag_datum
+	var/preference_type = ROLE_SPECIAL
+	var/protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Head of Personnel", "Chief Medical Officer", "Chief Engineer", "Research Director", "Captain", "Brig Physician")
```

Comment:
Shouldn't a list of protected jobs be per-special antag?

Reply:
I wanted to do this based off of the /datum/special_role however event_type is stored in the event holder as a datum rather than a reference to it and is initialised when the event is triggered, so not really sure how to do this one :/

Reply:
I'd like it to be done this way before merge. 

I don't really see how it is an issue by your description. Could you rephrase?

Reply:
```
//this datum is used by the events controller to dictate how it selects events
/datum/round_event_control
	var/name						//The human-readable name of the event
	var/typepath					//The typepath of the event datum /datum/round_event

	var/weight = 10					//The weight this event has in the random-selection process.
									//Higher weights are more likely to be picked.
									//10 is the default weight. 20 is twice more likely; 5 is half as likely as this default.
									//0 here does NOT disable the event, it just makes it extremely unlikely

	var/earliest_start = 20 MINUTES	//The earliest world.time that an event can start (round-duration in deciseconds) default: 20 mins
	var/min_players = 0				//The minimum amount of alive, non-AFK human players on server required to start the event.

	var/occurrences = 0				//How many times this event has occured
	var/max_occurrences = 20		//The maximum number of times this event can occur (naturally), it can still be forced.
									//By setting this to 0 you can effectively disable an event.

	var/holidayID = ""				//string which should be in the SSeventss.holidays list if you wish this event to be holiday-specific
									//anything with a (non-null) holidayID which does not match holiday, cannot run.
	var/wizardevent = FALSE
	var/random = FALSE				//If the event has occured randomly, or if it was forced by an admin or in-game occurance
	var/alert_observers = TRUE		//should we let the ghosts and admins know this event is firing
									//should be disabled on events that fire a lot

	var/list/gamemode_blacklist = list() // Event won't happen in these gamemodes
	var/list/gamemode_whitelist = list() // Event will happen ONLY in these gamemodes if not empty

	var/triggering	//admin cancellation
```
The typepath in the event controller is not instantiated, so the default values for the datum need to be set there again, I could add a check in the round_event_control which would make it easier, but would be changing other code, but I do that if it makes it easier.

Reply:
Do that, I'd like to see how it looks and you can always revert it

Reply:
Done @qwertyquerty